src/Orchard.Web/Modules/Orchard.DynamicForms/Views/EditorTemplates/FormBindings.cshtml

```diff
@@ -22,8 +22,8 @@
             <tbody>
                 @if (partDescriptor.BindingContexts.Any()) {
                     foreach (var partBinding in partDescriptor.BindingContexts.SelectMany(x => x.Bindings)) {
-                         var partBindingId = String.Format("binding-{0}-{1}", partDescriptor.Part.PartDefinition.Name.HtmlClassify(), partBinding.Name.HtmlClassify());
-                         var partBindingState = partState.Bindings.ElementAtOrDefault(partBindingIndex) ?? new BindingSettings();
+                        var partBindingId = String.Format("binding-{0}-{1}", partDescriptor.Part.PartDefinition.Name.HtmlClassify(), partBinding.Name.HtmlClassify());
+                        var partBindingState = partState.Bindings.FirstOrDefault(f => f.Name == partBinding.Name) ?? new BindingSettings();
```

Comment:
No it never is null because of this at the end of the line: ?? new BindingSettings()